mlir/lib/Dialect/Math/Transforms/LegalizeToF32.cpp

```diff
@@ -57,7 +57,9 @@ void mlir::math::populateLegalizeToF32TypeConverter(
   });
   typeConverter.addTargetMaterialization(
       [](OpBuilder &b, Type target, ValueRange input, Location loc) {
-        return b.create<arith::ExtFOp>(loc, target, input);
+        auto extFOp = b.create<arith::ExtFOp>(loc, target, input);
+        extFOp->setAttr("eliminatable", b.getBoolAttr(true));
```

Comment:
Setting "discardable" attributes in one pass and checking for them in another pass is generally unsafe. Is there a way to do this without attributes? Maybe an option for the pass that inserts the ops?

Reply:
Does is matter which pass produced the `truncf / extf` pairs? I would start with a pass that eliminates all of them. You can add an optional lambda to rewrite pattern that users can set as a kind of "filter function".

Reply:
It's according to the discussion in [PR](https://github.com/llvm/llvm-project/pull/92482), and the use case is provided [here](https://github.com/llvm/llvm-project/pull/92482#issuecomment-2127383611) by @krzysz00.

Reply:
> Does it matter which pass produced the `truncf / extf` pairs? I would start with a pass that eliminates all of them. You can add an optional lambda to rewrite pattern that users can set as a kind of "filter function".

It does not matter on my side, I would like to eliminate all of them as well. But we may need to include @krzysz00 into the discussion. BTW, do you have a PR right now?

Reply:
moved to `arith` dialect, and added optional filter function.